modular_citadel/code/game/machinery/cryopod.dm

```diff
@@ -0,0 +1,441 @@
+			continue//means we already moved whatever this thing was in
+			//I'm a professional, okay
+		for(var/T in preserve_items)
+			if(istype(W, T))
+				if(control_computer && control_computer.allow_items)
+					control_computer.frozen_items += W
+					mob_occupant.transferItemToLoc(W, control_computer, TRUE)
+				else
+					mob_occupant.transferItemToLoc(W, loc, TRUE)
+
+	for(var/obj/item/W in mob_occupant.GetAllContents())
+		qdel(W)//because we moved all items to preserve away
+		//and yes, this totally deletes their bodyparts one by one, I just couldn't bother
+
+	if(iscyborg(mob_occupant))
+		var/mob/living/silicon/robot/R = occupant
+		if(!istype(R)) return ..()
+
+		R.contents -= R.mmi
+		qdel(R.mmi)
+
+	// Ghost and delete the mob.
+	if(!mob_occupant.get_ghost(1))
+		if(world.time < 30 * 600)//before the 30 minute mark
+			mob_occupant.ghostize(0) // Players despawned too early may not re-enter the game
+		else
+			mob_occupant.ghostize(1)
+	QDEL_NULL(occupant)
+	open_machine()
+	name = initial(name)
+
+/obj/machinery/cryopod/MouseDrop_T(mob/living/target, mob/user)
+	if(!istype(target) || user.incapacitated() || !target.Adjacent(user) || !Adjacent(user) || !ismob(target) || (!ishuman(user) && !iscyborg(user)) || !istype(user.loc, /turf) || target.buckled)
+		return
+
+	if(occupant)
+		to_chat(user, "<span class='boldnotice'>The cryo pod is already occupied!</span>")
+		return
+
+	if(target.stat == DEAD)
+		to_chat(user, "<span class='notice'>Dead people can not be put into cryo.</span>")
+		return
+
+	if(target.client && user != target)
+		if(iscyborg(target))
+			to_chat(user, "<span class='danger'>You can't put [target] into [src]. They're online.</span>")
+		else
+			to_chat(user, "<span class='danger'>You can't put [target] into [src]. They're conscious.</span>")
+		return
+	else if(target.client)
+		if(alert(target,"Would you like to enter cryosleep?",,"Yes","No") == "No")
+			return
+
+	var/generic_plsnoleave_message = " Please adminhelp before leaving the round, even if there are no administrators online!"
+
+	if(target == user && world.time - target.client.cryo_warned > 5 * 600)//if we haven't warned them in the last 5 minutes
+		var/caught = FALSE
+		if(target.mind.assigned_role in GLOB.command_positions)
+			alert("<span class='userdanger'>You're a Head of Staff![generic_plsnoleave_message]</span>")
+			caught = TRUE
+		if(iscultist(target) || is_servant_of_ratvar(target))
+			to_chat(target, "<span class='userdanger'>You're a Cultist![generic_plsnoleave_message]</span>")
+			caught = TRUE
+		if(istype(SSticker.mode, /datum/antagonist/blob))
+			if(target.mind in GLOB.overminds)
+				alert("<span class='userdanger'>You're a Blob![generic_plsnoleave_message]</span>")
+				caught = TRUE
+		if(is_devil(target))
+			alert("<span class='userdanger'>You're a Devil![generic_plsnoleave_message]</span>")
+			caught = TRUE
+		if(istype(SSticker.mode, /datum/antagonist/rev))
```

Comment:
Test this to make sure it actually triggers during rev. I don't think SSticker.mode uses antag datums to determine gamemodes.

Reply:
It's what I saw most other code using, sooo.

This is also just purely a catch people thing as well, and it worked for cult mode, sooo.